python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -108,22 +107,32 @@ function reduceConsolidateAIConfig(
   action: MutateAIConfigAction,
   responseConfig: AIConfig
 ): ClientAIConfig {
+  // Make sure prompt structure is properly updated. Client input and metadata takes precedence
+  // since it may have been updated by the user while the request was in flight
+  const consolidatePrompt = (statePrompt: ClientPrompt) => {
+    const responsePrompt = responseConfig.prompts.find(
+      (resPrompt) => resPrompt.name === statePrompt.name
+    );
+    return {
+      ...responsePrompt,
+      ...statePrompt,
+      metadata: {
+        ...responsePrompt!.metadata,
+        ...statePrompt.metadata,
+      },
+    } as ClientPrompt;
+  };
+
   switch (action.type) {
-    case "ADD_PROMPT_AT_INDEX":
+    case "ADD_PROMPT_AT_INDEX": {
+      return reduceReplacePrompt(
+        state,
+        action.prompt._ui.id,
```

Comment:
Note that this action doesn't have an index explicitly passed in on dispatch, so need to grab it from the newly-inserted prompt in the action